RFC/src/RFC-0241_AtomicSwapXMR.md

```diff
@@ -0,0 +1,766 @@
+
+Please take note of the notation used in [TariScript] and specifically notation used on the signatures on the [transaction inputs](RFC-0201_TariScript.md#transaction-input-changes) and on the signatures on the [transaction outputs](RFC-0201_TariScript.md#transaction-output-changes), other notation will be noted in the [Notation](#notation) section.
+
+
+### TL;DR
+
+This scheme revolves around Alice, who wants to exchange some of her Tari for Bob's Monero. Because they don't 
+trust each other, they have to commit some values to do the exchange. And if something goes wrong here, we want to ensure
+that we can refund both parties either in Monero or Tari.
+
+How this works is that Alice and Bob create a shared output on both chains. The Monero output is a simple aggregate key
+to unlock the UTXO, while the Tari UTXO is unlocked by either one of three aggregate keys. The block height determines the unlock key on Tari. 
+
+The Tari aggregate keys are constructed so that the swap transaction's signature will reveal Bob's Monero key
+so that Alice has both keys, which allows her to claim the Monero. While the refund transaction's signature will reveal
+Alice's Monero key so that Bob has both keys, and he claims the Monero.
+
+To ensure that we can always claim the refund if Bob disappears after Alice posts the Tari UTXO, we need to
+ensure that this refund transaction is completed and signed by both Alice and Bob before Alice publishes the Tari UTXO.
+This ensures that in the case that Bob disappears, Alice can reclaim her Tari. And if Bob reappears, he can reclaim his Monero.
+
+But in the case where Alice disappears after Bob posts the Monero transaction, we need to create a lapse transaction for
+Bob to claim the Tari. This transaction is also completed and signed before the first Tari UTXO is published. This transaction
+will reveal Bob's Monero key so that if Alice reappears, she can claim the Monero. 
+
+![swap flow](assets/TXR_XMR_flow.png)
+
+### TariScript
+
+The Script used for the Tari UTXO is as follows:
+``` TariScript,ignore
+   CheckHeight(height_1)
+   LtZero
+   IFTHEN
+      PushPubkey(K_{Ss})
+   Else
+      CheckHeight(height_2)
+      LtZero
+      IFTHEN
+         PushPubkey(K_{Sr})
+      Else
+         PushPubkey(K_{Sl})
+      ENDIF
+   ENDIF
+```
+
+Here `height_1` is the lock height till Alice can claim the transaction. If Alice fails to publish the refund transaction
+after `height_2,` Bob can claim the lapse transaction.
+
+### Negotiation
+
+Alice and Bob have to negotiate about the exchange rate and the amount to be exchanged in the atomic swap. 
+They also need to decide how the two UTXO's will look on the blockchain. To accomplish this, the following needs to be finalized:
+
+* Amount of Tari to swap for the amount of Monero
+* Monero public key parts \\(X_a\\), \\(X_b\\) and its aggregate form \\(X\\)
+* Tari [script key] parts \\(K_{Ssa}\\), \\(K_{Ssb}\\) and its aggregate form \\(K_{Ss}\\)
```

Comment:
You use the subscripts s,r and l here but nowhere do you really describe what they are in the description you just gave. Sort of assuming it spend, refund and lapse but right here you can make it clear